vignettes/programming_strategy.Rmd

```diff
@@ -65,16 +65,53 @@ case-sensitive. Thus they must be in lower case.
 
 # Functions in R
 
-Functions in `{admiral}` must follow the modular approach, i.e. functions which add
-variables should add only a single variable except if other variables are
-closely related to the derived variable and can not be derived without it. For
-example `AAGE` (analysis age) and `AAGEU` (analysis age units) should be derived
-by the same function because it does not make sense to derive `AAGEU` without
-`AAGE`.
+## Function Design
 
-Variables like `BASE` (baseline value) and `BNRIND` (baseline reference range
-indicator) must be derived by two separate function calls as it is reasonable to
-derive only one of them.
+Firstly, it is important to explain how we decide on the need for new derivation functions. 
+
+If a derivation rule or algorithm is common and highly similar across different variables/parameters
+(e.g. study day or duration) then we would provide a generic function that can be used to satisfy all
+the times this may be needed across different ADaMs. Similarly, if we feel that a certain derivation
+could be useful beyond a single purpose we also would provide a generic function (e.g. instead of a
+last known alive date function, we have an extreme date function where a user could find the last date
+from a selection, or for example the first).
+
+Otherwise if we feel that a derivation rule is a unique need or sufficiently complex to justify then we
+opt for a dedicated function for that specific variable/parameter (e.g. treatment-emergent flag for AEs).
+
+If certain variables are closely connected (e.g. an imputed date and the corresponding imputation flag)
+then a single function would provide both variables.
+
+If something needed for ADaM could be achieved simply via an existing tidyverse function, then we do not
+wrap this into an admiral function, as that would add an unnecessary extra layer for users.
+
+The following principles are key when designing a new function:
```

Comment:
Should these line up with the the four principles in the README?

* Usability
* Simplicity
* Findability 
* Readability

Reply:
I guess I see them being the same.  The functions are the package and we have the 4 design principles that influence our practical approach to designing the functions.  Anyways, I am partial to short declarative sentences for designing stuff - as it is something I repeat to myself as I code things... e.g. "one proc away".  If you both are happy with how it reads, then I don't really have an issue.

Reply:
I see. Thanks Ben. We still have the whole "how to program a function, step by step" issue in the backlog. I think when we get to that we might need to adjust some wording as well :) I resolve this for now.